### PR TITLE
feat: Login sign up

### LINKED ISSUE #28: feat: Implement AuthFlow
* In the module, implement entry via login/registration process and also make a _Continue as a Guest_ option.
* As of now, we'll be considering firebase as a backend for our services, so make use of their docs and res.
* Try aligning UI as per the mockups provided

Ref: https://firebase.google.com/docs/auth/android/firebaseui

### LINKED ISSUE #32: feat: Firebase Authentication
We have to create a Login and SignUp Page linked with Firebase so that we can authenticate the users in the app and also take a count of the user indivisually!

### PR DESCRIPTION
**Pull request by raghavSharma1472**
- [x] Close #32 
- [x] Close #28

- First Steps towards #28 .
- Fixes ##32.

### Description
Mostly all the UI fixed for Login and Sign Up Page except a few
Basic Raw code for Login and Sign Up in firebase added

|Login|SignUp|
|--|--|
|Login|SignUp|

### CODE REVIEW

**Comment by s-ayush2903** [comment]
SignUp/Login doesn't work, will push a fix

File: app/src/main/res/layout/activity_firstscreen.xml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        android:layout_height="wrap_content"
+        android:layout_marginTop="8dp"
+        android:layout_marginBottom="369dp"
+        android:ems="10"
+        android:inputType="textPersonName"
+        android:text="Password"
+        android:textColor="@color/dotInactive"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintStart_toStartOf="@+id/emailId"
+        app:layout_constraintTop_toBottomOf="@+id/emailId" />
```
Comment (s-ayush2903) [inline app/src/main/res/layout/activity_firstscreen.xml]: @raghavSharma1472 Just to point it out(as it is quite necessary), that we use a `+` sign only while _creating_ new ids. If we use it the plus sign for referring ids(which we've done in this case) then the compiler won't be able to figure out where that particular ID refers to as there would be many implementations of the same. So for referring views use: `@id/view_id_name` not `@+id/view_id_name`

File: app/src/main/res/layout/activity_firstscreen.xml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/password" />
+
+    <Button
+        android:id="@+id/signup_button"
+        android:layout_width="210dp"
+        android:layout_height="wrap_content"
+        android:layout_marginBottom="22dp"
```
Comment (s-ayush2903) [inline app/src/main/res/layout/activity_firstscreen.xml]: Also we don't need margin params for each view to be associated to the screen size, better we'll be constraining them via providing margins from pre-existing views that'll make the layout dynamic and responsive.

**Review by s-ayush2903 — COMMENTED** [review]
Work Left:
* Converting newly added activities to fragments
* Handling nav among them
* Maintaining a logged-in flag
* Converting EditTexts to [TILs](https://material.io/components/text-fields/android#using-text-fields)

**Review by s-ayush2903 — COMMENTED** [review]
Current Status:
- [x] Converting newly added activities to fragments(Login & Signup are now Frags)
- [x] Handling nav among them(Tightly integrated Jetpack Nav)
- [x] Maintaining a logged-in session pref
-------
Work Left:
- [ ] Converting EditTexts to [TILs](https://material.io/components/text-fields/android#using-text-fields)
- [ ] Using material buttons
- [ ] Creating a Database in firebase console
- [ ] Set the login functionality working
- [ ] Set up email verification(optional)

**Comment by s-ayush2903** [comment]
Log-in works now. I think there is an issue with the material design dep, when implemented changed the color of bottomNav to black, so reverted it. Mentioning just to keep a note

File: app/src/main/java/org/locer/in/LoginFragment.kt [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import org.locer.`in`.LoginFragmentDirections.Companion.actionLoginFragmentToHomeFragment
+import org.locer.`in`.databinding.FragmentLoginBinding
+
+private const val TAG = "LoginFragment"
+
+class LoginFragment : Fragment() {
+
+    private lateinit var loginFragmentBinding: FragmentLoginBinding
+    private val sharedPreferenceUtil: SharedPreferenceUtil by
+    lazy { SharedPreferenceUtil(requireContext()) }
```
Comment (s-ayush2903) [inline app/src/main/java/org/locer/in/LoginFragment.kt:27]: We should extract this common functionality out of these two classes in future

**Review by s-ayush2903 — APPROVED** [review]

**Comment by s-ayush2903** [comment]
Will merge on green build

File: app/src/main/res/layout/activity_firstscreen.xml (hunk 2 above)
Comment (raghavSharma1472) [reply app/src/main/res/layout/activity_firstscreen.xml]: ok

File: app/src/main/res/layout/activity_firstscreen.xml (hunk 1 above)
Comment (raghavSharma1472) [reply app/src/main/res/layout/activity_firstscreen.xml]: ok

**Pull request merged**